### PR TITLE
WalletConnect and general code improvements (version 1.1)

### LINKED ISSUE #160: Sort vaults list by usd value
Show vaults with 0 in deposit at the bottom of the list @roienatan

### PR DESCRIPTION
**Pull request by roienatan**
resolves: #109 
resolves: #86
resolves: #123 
resolves #160
resolves: #151
resolves: #170 
resolves: #161 
resolves: #163  
_Also many general code improvements and changes_

### CODE REVIEW

File: src/components/Header.tsx [hunk 1]
```diff
@@ -51,9 +53,39 @@ export default function Header() {
...
+            console.log("accountsChanged", accounts);
+            dispatch(updateSelectedAddress(accounts[0]));
+          });
+          actualProvider.on("chainChanged", (chainId) => {
+            // Handle the new chain.
+            // Correctly handling chain changes can be complicated.
+            // We recommend reloading the page unless you have good reason not to.
+            window.location.reload();
+          });
+          // if (provider instanceof WalletConnectProvider) {
```
Comment (roienatan) [inline src/components/Header.tsx:79]: @leviyehonatan Please remove comments

File: src/actions/index.ts [hunk 2]
```diff
@@ -18,6 +18,11 @@ export const updateWalletBalance = (ethBalance: any, hatsBalance: any) => ({
   hatsBalance
 })
 
+export const updateNetwork = (network: string) => ({
```
Comment (roienatan) [inline src/actions/index.ts:21]: It's necessary to have the current `NETWORK` in the redux? instead of just checking it using `getNetworkNameByChainId`

**Review by roienatan — COMMENTED** [review]
@leviyehonatan Submit vulnerabilities vaults are absent. Please fix.

File: src/actions/index.ts (hunk 2 above)
Comment (leviyehonatan) [reply src/actions/index.ts:21]: misunderstanding

**Comment by leviyehonatan** [comment]
resolves #160

File: src/components/DepositWithdraw/DepositWithdraw.tsx [hunk 3]
```diff
@@ -103,6 +99,7 @@ export default function DepositWithdraw(props: IProps) {
   const [availableToWithdraw, setAvailableToWithdraw] = useState(BigNumber.from(0));
   const [withdrawAmount, setWithdrawAmount] = useState(BigNumber.from(0));
   const [userShares, setUserShares] = useState(BigNumber.from(0));
+  const apy = hatsPrice ? calculateApy(props.data.parentVault, hatsPrice, tokenPrice) : 0;
 
   useEffect(() => {
```
Comment (leviyehonatan) [inline src/components/DepositWithdraw/DepositWithdraw.tsx:104]: i am not sure about the need for this data, @roienatan please let me know why this data is used and why there is polling here

Comment (shayzluf) [reply src/components/DepositWithdraw/DepositWithdraw.tsx:104]: apy is the incentive mechanism. it should be present as we kick off the lm incentives

File: src/components/Airdrop/components/Airdrop/Airdrop.tsx [hunk 4]
```diff
@@ -9,34 +8,32 @@ import { t } from "i18next";
...
+  const [userInput, setUserInput] = useState(walletAddress);
   const nftET = useSelector((state: RootState) => state.dataReducer.airdrop?.nft);
   const tokenET = useSelector((state: RootState) => state.dataReducer.airdrop?.token);
   const [nftEligibilityStatus, setNFTEligibilityStatus] = useState(EligibilityStatus.UNKNOWN);
   const [tokenId, setTokenId] = useState<string>();
-  const [tokenEligibilityStatus, setTokenEligibilityStatus] = useState(EligibilityStatus.UNKNOWN);
-  const [tokenAmount, setTokenAmount] = useState<number>();
+  //const [tokenEligibilityStatus, setTokenEligibilityStatus] = useState(EligibilityStatus.UNKNOWN);
+  //const [tokenAmount, setTokenAmount] = useState<number>();
   const [inTokenAirdop, setInTokenAirdrop] = useState(false);
```
Comment (leviyehonatan) [inline src/components/Airdrop/components/Airdrop/Airdrop.tsx:28]: is this state needed?

File: src/components/Airdrop/components/Airdrop/Airdrop.tsx [hunk 5]
```diff
@@ -9,34 +8,32 @@ import { t } from "i18next";
...
-        const tokenId = Object.keys(nftET!).find(key => nftET![key] === address);
-        setTokenId(tokenId!);
-        if (await isRedeemed(tokenId!, address)) {
+  const checkAirdrop = useCallback(async () => {
+    if (userInput && isAddress(userInput) && nftET) {
+      const address = normalizeAddress(userInput);
+      if (Object.values(nftET).includes(address)) {
+        const tokenId = Object.keys(nftET!).find(key => nftET[key] === address);
+        setTokenId(tokenId);
+        if (tokenId && await isRedeemed(tokenId, address)) {
```
Comment (leviyehonatan) [inline src/components/Airdrop/components/Airdrop/Airdrop.tsx:36]: its better to read the contract as a hook as in useCall in DepositWithdraw, the actual definition of the contract is defined in contract hooks file

File: src/components/Airdrop/components/Airdrop/Airdrop.tsx [hunk 6]
```diff
@@ -9,34 +8,32 @@ import { t } from "i18next";
...
 
 export default function Airdrop() {
-  const [userInput, setUserInput] = useState("");
-  const [pendingWallet, setPendingWallet] = useState(false);
+  const { hasClaimed, isRedeemed } = useActions();
+  const { walletAddress } = useParams();
+  const [userInput, setUserInput] = useState(walletAddress);
   const nftET = useSelector((state: RootState) => state.dataReducer.airdrop?.nft);
   const tokenET = useSelector((state: RootState) => state.dataReducer.airdrop?.token);
   const [nftEligibilityStatus, setNFTEligibilityStatus] = useState(EligibilityStatus.UNKNOWN);
```
Comment (leviyehonatan) [inline src/components/Airdrop/components/Airdrop/Airdrop.tsx:24]: i think this state is not need as it is calculated once we read from the contract

Comment (roienatan) [reply src/components/Airdrop/components/Airdrop/Airdrop.tsx:24]: @leviyehonatan ok I'm sure this code can be improved but that's for a separate PR. We first need to merge this.

**Comment by roienatan** [comment]
As discussed with @shayzluf "Claim" functionality will be tested later on.

**Review by leviyehonatan — APPROVED** [review]
this PR has been therally discussed and the further needed improvements will be taken care of on a different PR. approving

**Pull request merged**